cmd/build/main.go

```diff
@@ -204,32 +82,65 @@ func run() error {
 		return fmt.Errorf("invalid image type %q for distro %q and arch %q: %w", imgTypeName, distroName, archName, err)
 	}
 
-	// get repositories
-	repos, err := loadRepos(repositories, distroName, archName)
-	if err != nil {
-		return fmt.Errorf("failed to get repositories for %s/%s: %w", distroName, archName, err)
+	var reporeg *reporegistry.RepoRegistry
+	var customRepos []rpmmd.RepoConfig
+	if st, err := os.Stat(repositories); err == nil && st.Mode().IsRegular() {
+		repoConfig, err := rpmmd.LoadRepositoriesFromFile(repositories)
+		if err != nil {
+			return fmt.Errorf("failed to load repositories from %q: %w", repositories, err)
+		}
+		customRepos = repoConfig[archName]
+	} else {
+		// HACK: reporegistry hardcodes adding the "repositories"
+		// dir to the "repoConfigPaths" but the interface of
+		// "images" does not expect this.
+		// XXX: should we fix this in reporegistry?
+		repositories = filepath.Join(repositories, "..")
+		reporeg, err = reporegistry.New([]string{repositories})
+		if err != nil {
+			return fmt.Errorf("failed to load repositories from %q: %w", repositories, err)
+		}
 	}
```

Comment:
I'm not sure I like that depending on the `repositories` being a regular file (BTW, it also could be a symlink, no?) the code either directly loads the repo config or constructs the RepoRegistry.

I'd suggest unifying how we call `manifestgen.New()` in both cases. IOW either always load the repo file (the distro is known, so this can also be done in case `repositories` is a directory) or always construct `RepoRegistry` (if `repositories` is a file, then load the parent directory. However, the behavior may be unexpected if the provided filename does not conform to `<distro>.json` naming scheme).

But I can live with this version as well...

Reply:
Yeah, excellent point. I will think about this, part of the issue is exactly what you point out, "build" currently supports pointing to arbitrarily named files that most likely not conform to <distro>.json - but I can rework this or (see above) we just keep "build" as is and only do a minimal port to manifestgen.

I'm not sure we have good options here though, moving single-file loading into reporegistry itself leaves the open question to what distro the repo should be attached (and it goes against the spirit of the package). Chaning OverrideReos to a path feels also awkward (and limits how this can be used, one could imagine an override via e.g. a network request). Ideas very welcome, I would love to see this improved.

Reply:
I think dropping the forced `repositories` path from `LoadAllRepositories()` in reporegistry is fine, but let's make a big deal out of it and update osbuild-composer immediately so we don't forget and end up scratching our heads next time we update the dependency.

Reply:
Let's keep it as is for now...